### DISCUSSION TITLE
Clarify use of scope option instead of array

### DISCUSSION THREAD

**Discussion by afdev82** [fnando/i18n-js#648] [Q&A]
Hi,

I had the following working code to get a translation:

```
message = I18n.t(['configurations.jobs.statuses.pending', status.config.last_job], {
  defaults: [{ scope: 'configurations.jobs.statuses.pending.default' }],
  job: jobName,
  status: status.config.job_status
});
```

Then I wanted to use always the scope option instead of interpolating strings or use the array syntax.
So I wrote it like this:

```
message = I18n.t(status.config.last_job, {
  defaults: [{ scope: 'configurations.jobs.statuses.pending.default' }],
  job: jobName,
  scope: 'configurations.jobs.statuses.pending',
  status: status.config.job_status
});
```

In case of a missing translation, the first code returns the translation using the defaults ('configurations.jobs.statuses.pending.default'), but the second code doesn't consider it and it returns the "missing translation" message instead.
What am I doing wrong here? I thought the scope option could be used instead of the array syntax to specify the scope, e.g.:

```
message = I18n.t(status.config.last_job, {
  scope: 'configurations.jobs.statuses.pending',
}
```

instead of

```
message = I18n.t(['configurations.jobs.statuses.pending', status.config.last_job])
```

Thank you for your support!

**Comment by PikachuEXE** [comment]
I think https://github.com/fnando/i18n-js/blob/v3/spec/js/translate.spec.js provides the test cases about scope option.
Is there any test case there (or in other spec files) describe your use case?
If not can you try to fork, add the test case, run locally & report back result?
I can take a look at fixing it when it can reproduced via test case(s).

**Reply by afdev82** [reply]
Thank you for your reply.

Yes, I will have a look at the specs and add a test case if needed.
My question was if I'm doing something wrong and the options are not supposed to work in this way.
For that reason I asked a question and I haven't open an issue for a bug report.

**Reply by afdev82** [reply]
Done, thank you.

**Reply by PikachuEXE** [reply]
I have no idea if that's expected behaviour or a bug neither.
But having a test case making things easier to understand.

**Reply by PikachuEXE** [reply]
OK the CI run and failed for this test case: (Which is now much easier to be understood by me)
```js
I18n.translate("monster", {
  defaults: [{ scope: "greetings.stranger" }],
  scope: "greetings"
});
```

I would expect the library is doing the following:
- scope: `greetings` & key: `monster` > lookup `en.greetings.monster` > missing
- try first default > scope: `greetings.stranger` & key: `monster` > lookup `en.greetings.stranger.monster` > missing
- return `[missing xxx translation]`

What do you think?

**Reply by afdev82** [reply]
I thought the two examples should be equivalent:

```
I18n.translate(["greetings", "monster"], {
  defaults: [{ scope: "greetings.stranger" }]
});
```
and
```
I18n.translate("monster", {
  defaults: [{ scope: "greetings.stranger" }],
  scope: "greetings"
});
```

Without the defaults option, they have the same behavior.
I would expect that:

 -  scope: greetings & key: monster > lookup en.greetings.monster > missing
 -  try first default > scope: greetings.stranger > lookup en.greetings.stranger > found
 -  return "Hello stranger!"

It's the same behavior of the [first](https://github.com/adnotam/i18n-js/blob/scope_with_defaults/spec/js/translate.spec.js#L153) and the [second](https://github.com/adnotam/i18n-js/blob/scope_with_defaults/spec/js/translate.spec.js#L158) example, I don't know why adding the scope option instead of the array syntax to specify the key should change that.
The first example doesn't use the key "Hello!" with the default scope, it just returns it.

```
I18n.translate("Hello!", {defaults: [{scope: "greetings.stranger"}]});
```

**Reply by PikachuEXE** [reply]
I think the `scope` in option might or might not have the same meaning as `scope` in `defaults`
And the question for me would be "if the option `scope` (e.g. `greetings`) should be applied to the `scope` in `defaults"

In this case I have no idea
@fnando ?
